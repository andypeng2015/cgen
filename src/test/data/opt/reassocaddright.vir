module foo

export function w $foo(w %x, w %y, w %z) {
@start:
  %r = add.w %z, 1_w
  %r = add.w %y, %r
  %r = add.w %x, %r
  ret %r
}
