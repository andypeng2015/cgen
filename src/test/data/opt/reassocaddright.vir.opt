module foo

export function w $foo(w %x, w %y, w %z) {
@2:
  %4 = add.w %z, 0x1_w ; @10
  %3 = add.w %x, %y ; @9
  %5 = add.w %3, %4 ; @11
  ret %5
}
