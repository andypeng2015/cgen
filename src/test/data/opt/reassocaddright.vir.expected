module foo

export function w $foo(w %x, w %y, w %z) {
@0:
  %1 = add.w %z, 0x1_w ; @5
  %0 = add.w %x, %y ; @4
  %2 = add.w %0, %1 ; @6
  ret %2
}
